### PR TITLE
feat: Qdrant enhancement

### LINKED ISSUE #989: [Feature Request] Qdrant support vector delete by key word
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

currently we only support deleting by using id, it's hard to track, better support with deleting by key word

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by koch3092**
## Description
refactor qdrant module and add some features

- Added class QdrantManager to manage global collections' CRUD operations.
- Added an explicit QdrantStorage.close_client() method to prevent concurrency conflicts in local storage access.
- Added QdrantStorage.update(), which allows updating the vector itself or updating the payload.
- Added filter functionality to QdrantStorage.query(), enabling searches using QdrantClient’s native Filter support.

## Motivation and Context

close #989 

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/storages/vectordb_storages/qdrant.py [hunk 1]
```diff
@@ -273,38 +420,134 @@ def add(
...
             )
 
+    def update(
+        self,
+        records: List[VectorRecord],
+        only_payload: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        r"""
+        Updates the payload of the vectors in the collection.
```
Comment (lightaime) [inline camel/storages/vectordb_storages/qdrant.py:430]:
```suggestion
        r"""Updates the payload of the vectors in the collection.
```

File: camel/storages/vectordb_storages/qdrant.py [hunk 2]
```diff
@@ -273,38 +420,134 @@ def add(
...
+            if op_info.status != UpdateStatus.COMPLETED:
+                raise RuntimeError(
+                    "Failed to delete vectors in Qdrant, operation info: "
+                    f"{op_info}"
+                )
 
     def status(self) -> VectorDBStatus:
-        status = self._get_collection_info(self.collection_name)
+        r"""
+        Returns the status of the collection.
```
Comment (lightaime) [inline camel/storages/vectordb_storages/qdrant.py:543]:
```suggestion
        r"""Returns the status of the collection.
```

File: camel/storages/vectordb_storages/qdrant.py [hunk 3]
```diff
@@ -273,38 +420,134 @@ def add(
...
+        r"""Deletes points from the collection based on either IDs or payload
+            filters.
 
         Args:
             ids (List[str]): List of unique identifiers for the vectors to be
                 deleted.
+            payload_filter (Optional[Dict[str, Any]], optional): A filter for
+                the payload to delete points matching specific conditions. If
+                `ids` is provided, `payload_filter` will be ignored unless both
+                are combined explicitly.
```
Comment (lightaime) [inline camel/storages/vectordb_storages/qdrant.py:488]: add some examples about how to use the `payload_filter` in docstring.

File: test/storages/vector_storages/test_qdrant.py [hunk 4]
```diff
@@ -96,3 +105,135 @@ def test_existing_collection():
...
+    manager = QdrantManager(path=tmpdir)
+
+    if manager.collection_exists(collection_name):
+        manager.delete_collection(collection_name)
+
+    storage = QdrantStorage(
+        vector_dim=4,
+        path=tmpdir,
+        collection_name=collection_name,
+    )
```
Comment (lightaime) [inline test/storages/vector_storages/test_qdrant.py:122]: What use cases we need to first define `QdrantManager ` and then `QdrantStorage`?

File: camel/storages/vectordb_storages/qdrant.py [hunk 5]
```diff
@@ -24,9 +24,178 @@
 from camel.types import VectorDistance
 from camel.utils import dependencies_required
 
+if TYPE_CHECKING:
+    from qdrant_client import QdrantClient
+
 _qdrant_local_client_map: Dict[str, Tuple[Any, int]] = {}
 
 
+class QdrantManager:
```
Comment (lightaime) [inline camel/storages/vectordb_storages/qdrant.py:33]: What use cases we must need to have `QdrantManager` separately? What if we just put all things into `QdrantStorage`?

**Review by lightaime — COMMENTED** [review]
Looks good but what do we must have `QdrantManager`? Can you explain?

**Comment by koch3092** [comment]
> Looks good but what do we must have `QdrantManager`? Can you explain?

Thanks @lightaime 

`QdrantManager` is designed to provide users with the ability to manage collections so that users can control the collections in their qdrant database in their business logic.

The function of `QdrantStorage` is CRUD for records in a single collection, such as `add() → add a record into collection`. Its processing scope should only be collection.

In the original design, `QdrantStorage` had private functions to operate collections, such as `_create_collection()` and `_delete_collection`. If they were directly converted to public functions, there would be security risks.

**Review by Wendong-Fan — COMMENTED** [review]
Hey @koch3092 , from Qdrant's documentation, I suggest we use client to manage both collection and vector, I feel there's no need to add one more layer to make the design complex
cc @lightaime

**Comment by koch3092** [comment]
Refactored association #1147

**Pull request closed without merging**